File: LR2/11.rb
# Считываем ввод и разделяем на части
input = gets.chomp.split

# Проверяем, что введено три элемента: N, M и S
if input.size < 3
  puts "Invalid input"
  exit
end

n_str, m_str, s = input[0], input[1], input[2]

begin
  # Преобразуем N и M в целые числа
  n = Integer(n_str)
  m = Integer(m_str)
rescue ArgumentError
  # Если преобразование не удалось, выводим ошибку
  puts "Invalid input"
  exit
end

# Проверяем, что размеры доски корректны
if n < 1 || m < 1
  puts "Invalid input"
  exit
end

# Проверяем, что строка S содержит только допустимые символы
unless s.chars.all? { |c| %w[L R U D].include?(c) }
  puts "Invalid input"
  exit
end

# Инициализируем счетчики и минимальные координаты
l = 0
r = 0
u = 0
d = 0
min_x = 1
min_y = 1

# Обрабатываем каждую команду
s.each_char do |c|
  case c
  when 'L'
    l += 1
    min_x += 1 if min_x - l < 1
    r -= 1 if r > 0
  when 'R'
    r += 1
    l -= 1 if l > 0
  when 'U'
    u += 1
    min_y += 1 if min_y - u < 1
    d -= 1 if d > 0
  when 'D'
    d += 1
    u -= 1 if u > 0
  end
end

# Проверяем существование допустимой позиции
if min_x > m || min_y > n
  puts "No"
else
  # Выводим минимальные координаты
  puts "(#{min_x},#{min_y})"
end

File: LR2/11.py
try:
    # Чтение и разбиение входных данных на части
    parts = input().split()
    if len(parts) != 3:
        print("Invalid input")
        exit()
    N_str, M_str, S = parts
    N = int(N_str)
    M = int(M_str)
except ValueError:
    # Обработка ошибок преобразования строк в числа
    print("Invalid input")
    exit()

# Проверка корректности размеров доски
if N < 1 or M < 1:
    print("Invalid input")
    exit()

# Проверка допустимости символов в строке команд
for c in S:
    if c not in {'L', 'R', 'U', 'D'}:
        print("Invalid input")
        exit()

# Инициализация минимальных координат и счетчиков команд
min_x = 1
min_y = 1
L = 0
R = 0
U = 0
D = 0

# Обработка каждой команды с динамическим обновлением координат
for c in S:
    if c == 'L':
        L += 1
        if min_x - L < 1:
            min_x += 1
        if R > 0:
            R -= 1
    elif c == 'R':
        R += 1
        if L > 0:
            L -= 1
    elif c == 'U':
        U += 1
        if min_y - U < 1:
            min_y += 1
        if D > 0:
            D -= 1
    elif c == 'D':
        D += 1
        if U > 0:
            U -= 1

# Проверка существования допустимой начальной позиции
if min_x > M or min_y > N:
    print("No")
else:
    print(f"({min_x},{min_y})")